src/common/services/password.svc.js

```diff
@@ -0,0 +1,75 @@
+import dataStore from "../data-store";
+import messagingSvc from "./messaging.svc";
+
+/**
+ * Configure the password that will be used to secure this account and enrypt the data.
+ * @param {string} password original password
+ * @param {string} confirmedPassword confirmed passord
+ */
+async function init(password, confirmedPassword) {
+  if (password !== confirmedPassword) {
+    throw new Error("Cannot initialzie. Passwords do not match!");
+  }
+  const storage = dataStore(password);
+  await storage.set("isPasswordSet", true);
+}
+
+/**
+ * Check if the provided password is the correct one, that is the one used to initalize the account.
+ * @param {string} password the passord
+ * @returns true if the passord is correct, false otherwise
+ */
+async function checkPassword(password) {
+  if (password && password.length) {
+    return _checkUserPassword(password);
+  }
+  return _checkCachedPassword();
+}
+
+/**
+ * Check if the user has already configured a password for the extension.
+ * @returns true if the user has already configured a password, false otherwise
+ */
+async function isInitialized() {
+  const storage = dataStore();
+  return storage.isInitialized();
+}
+
+/**
+ * Check if the user has provided a valid password since the extesion was last time loaded.
+ * @returns true if the user has already provided a valid password, false otherwise
+ */
+async function isUnlocked() {
+  const storage = dataStore();
+  return storage.isUnlocked();
+}
+
+async function _checkUserPassword(password) {
+  try {
+    const storage = dataStore(password);
+    const isPasswordSet = await storage.get("isPasswordSet");
+    return isPasswordSet === true;
+  } catch (err) {
+    console.log(err);
+    return false;
+  }
+}
+
+async function _checkCachedPassword() {
+  return new Promise(async (resolve) => {
+    await messagingSvc.onMessage("cached-password", async (msg) => {
```

Comment:
I think this listener would never get removed and it will also receive all future messages? is that correct?

Reply:
and you introduce this pattern to receive a reply:

1. add a listener 
2. send a message 
3. handler of the sent message, in return sends a message
4. listener receives the message of the handler

Do I understand this correct?

Reply:
another thought: I'd personally appreciate removing the `svc` duplication in the filename, since we have `services` already in the path.